internal/broker/data.go

```diff
@@ -0,0 +1,58 @@
+package broker
+
+import (
+	"bytes"
+	"encoding/json"
+	"text/template"
+
+	"github.com/ubuntu/oidc-broker/internal/providers/group"
+)
+
+// userInfo represents the user information that is returned to authd.
+type userInfo struct {
+	Name   string
```

Comment:
As we don’t use json reflection here and do serialization ourself, there is no need for those fields to be public, isn’t it?

Reply:
I see you are using the private fields in tests, but let’s not tamper production code for tests. In that case, it’s better to transcribed to another struct which has the mapping with public fields in `export_test.go` or just dump it to json in golden IMHO

Reply:
Well, the struct itself is already private. IMHO, changing its fields to private creates more nuisance than it helps prevent, no? I was trying to find some guidelines for idiomatic struct declarations and so on, but did not find any. Do you have some links?

Also, we do unmarshal this type quite a lot inside our code (when loading the cached info for the user and so on), so I'm not 100% sure we can make these private anyway.

Reply:
As told, this was relevant when we used templating, but not anymore with the new approach.